src/database.ts

```diff
@@ -3242,42 +3243,38 @@ class Database extends common.GrpcServiceObject {
         }
 
         const release = () => {
-          span.end();
           this.pool_.release(session!);
         };
 
         const runner = new TransactionRunner(
           session!,
           transaction!,
-          (err, resp) => {
+          async (err, resp) => {
             if (err) {
               setSpanError(span, err!);
             }
-            span.end();
-            runFn!(err, resp);
+            await runFn!(err, resp);
           },
           options
         );
 
-        runner.run().then(release, err => {
-          if (err) {
-            setSpanError(span, err!);
-          }
+        await runner.run().then(release, async err => {
```

Comment:
we are already doing .then() on runner.run(), why do we then need to await here?